content/300-guides/100-prisma-guides/100-prisma-client-transactions-guide.mdx

```diff
@@ -0,0 +1,773 @@
+
+#### When to use the `$transaction` API
+
+Consider the `$transaction` API if:
+
+* ✔ You want to update a batch that includes different types of records, such as emails and users. The records do not need to be related in any way.
+* ✔ You want to batch raw SQL queries (`$executeRaw`) - for example, for features that Prisma Client does not yet support.
+
+#### Scenario: Privacy legislation
+
+GDPR and other privacy legislation give users the right to request that an organization deletes all of their personal data. In the following example schema, a `User` can have many posts and private messages:
+
+```prisma
+model User {
+  id                Int       @id @default(autoincrement())
+  posts             Post[]
+  privateMessages   PrivateMessage[]
+}
+
+model Post {
+  id         Int     @id @default(autoincrement())
+  user       User    @relation(fields: [userId], references: [id])
+  userId     Int  
+  title      String
+  content    String
+}
+
+model PrivateMessage {
+  id         Int     @id @default(autoincrement())
+  user       User    @relation(fields: [userId], references: [id])
+  userId     Int  
+  message      String
+}
+```
+
+If a user invokes the right to be forgotten, we must delete three records: the user record, private messages, and posts. It is critical that _all_ delete operations succeed together or not at all, which makes this a use case for a transaction. However, using a single bulk operation like `deleteMany` is not possible in this scenario because we need to delete across three models. Instead, we can use the `$transaction` API to run three operations together - two `deleteMany` and one `delete`:
+
+```ts
+const id = 9; // User to be deleted
+
+const deletePosts = prisma.post.deleteMany({
+  where: {
+    userId: id
+  }
+})
+
+const deleteMessages = prisma.privatemessage.deleteMany({
+  where: {
+    userId: id
+  }
+})
+
+const deleteUser = prisma.post.delete({
+  where: {
+    id: id
+  }
+})
+
+await prisma.$transaction([deletePosts, deleteMessages, deleteUser]) // Operations succeed or fail together
+```
+
+#### Scenario: Pre-computed IDs and the `$transaction` API
+
+Dependent writes are not supported by the `$transaction` API - if operation A relies on the ID generated by operation B, use [nested writes](#nested-writes). However, if you _pre-computed_ IDs (for example, by generating GUIDs), your writes become independent. Consider the sign-up flow from the nested writes example:
+
+```ts
+await client.team.create({
+  data: {
+    name: 'Aurora Adventures',
+    members: {
+      create: {
+        email: 'alice@prisma.io',
+      },
+    },
+  },
+})
+```
+
+Instead of auto-generating IDs, change the `id` fields of `Team` and `User` to a `String` (if you do not provide a value, a UUID is generated automatically). We recommend using UUIDs:
```

Comment:
A minor note: UUIDs are sometimes challenged versus CUIDs etc. Maybe it's fine to leave that as something we recommend out of simplicity, but this may be something people challenge.